File: Mini_Games/Guess_The_Number/GTN.py
import random

Player = False

while Player == False:  
    # Allow user select number range
    try:
        print("What range? Please enter 2 numbers separately ")
        num_start = int(input("Start range: "))
        num_end = int(input("End range: "))
    except ValueError:
        print("Invalid input. Please enter a valid number.")
        continue  
    
    number = random.randrange(num_start, num_end)

    while True:
        # Player guessing
        try:
            user_option = int(input("Your guess: "))
        except ValueError:
            print("Invalid input. Please enter a valid number.")
            continue  

        # Possible outcomes
        if user_option == number:
            print("You got it! ")
            break
        elif user_option > number:
            print("Lower")
            
        elif user_option < number:
            print("Higher")
        else:   
            print("Invalid option. Please choose a number")
            continue

    # Section to start/end game
    play = input("\nPlay again? Yes or No: ").lower()
    if play == "yes":
        Player = False
    else:
        print("Thanks for playing!")
        break       

File: Games_Website/all-js/GTN.js
let numberToGuess;
let gameSection = document.getElementById("GTN-game-section");
let feedback = document.getElementById("GTN-feedback");
let playAgainButton = document.getElementById("GTN-play-again");

function startGame() {
    let startRange = parseInt(document.getElementById("GTN-start-range").value);
    let endRange = parseInt(document.getElementById("GTN-end-range").value);

    if (isNaN(startRange) || isNaN(endRange) || startRange >= endRange) {
        alert("Please enter valid numbers for the range.");
        return;
    }

    numberToGuess = Math.floor(Math.random() * (endRange - startRange + 1)) + startRange;
    gameSection.style.display = "block";
    playAgainButton.style.display = "none";
    feedback.textContent = "";
}

function makeGuess() {
    let userGuess = parseInt(document.getElementById("GTN-user-guess").value);

    if (isNaN(userGuess)) {
        feedback.textContent = "Invalid input. Please enter a valid number.";
        return;
    }

    if (userGuess === numberToGuess) {
        feedback.textContent = "You got it!";
        playAgainButton.style.display = "block";
    } else if (userGuess > numberToGuess) {
        feedback.textContent = "The number is Lower than " + document.getElementById("GTN-user-guess").value;
        document.getElementById("GTN-user-guess").value = "";
    } else {
        feedback.textContent = "The number is Higher than " + document.getElementById("GTN-user-guess").value;
        document.getElementById("GTN-user-guess").value = "";
    }
}

function restartGame() {
    document.getElementById("GTN-start-range").value = "";
    document.getElementById("GTN-end-range").value = "";
    document.getElementById("GTN-user-guess").value = "";
    gameSection.style.display = "none";
    playAgainButton.style.display = "none";
    feedback.textContent = "";
}